R/module_filter_manager.R

```diff
@@ -137,22 +137,29 @@ filter_manager_srv <- function(id, filtered_data_list, filter) {
       }
 
     # Create mapping fo filters to modules in matrix form (presented as data.frame).
+    # Modules get NAs for filteres that cannot be set for them.
     mapping_matrix <- reactive({
-      module_states <- lapply(filtered_data_list, function(x) x$get_filter_state())
-      mapping_ragged <- lapply(module_states, function(x) vapply(x, `[[`, character(1L), "id"))
-      all_names <- vapply(slices_global(), `[[`, character(1L), "id")
-      mapping_smooth <- lapply(mapping_ragged, is.element, el = all_names)
-      as.data.frame(mapping_smooth, row.names = all_names, check.names = FALSE)
+      state_ids_global <- vapply(slices_global(), `[[`, character(1L), "id")
+      mapping_smooth <- lapply(filtered_data_list, function(x) {
+        state_ids_local <- vapply(x$get_filter_state(), `[[`, character(1L), "id")
+        state_ids_allowed <- vapply(x$get_available_teal_slices()(), `[[`, character(1L), "id")
```

Comment:
I think we can do slight improvement in the related code. Line 208 in this file is not needed as `FilteredData$set_available_teal_slices` subsets by available datasets already.

https://github.com/insightsengineering/teal/blob/2cb0011d7f47dce7abc648e1840a4b561b6c07da/R/module_filter_manager.R#L208

https://github.com/insightsengineering/teal.slice/blob/57d873c0b06893e556f221a9829820395d09ca6e/R/FilteredData.R#L153